controllers/terminal_controller.go

```diff
@@ -1375,21 +1443,25 @@ func NewClientSetFromSecret(ctx context.Context, config *rest.Config, secret *co
 	}
 
 	if token, ok := secret.Data[corev1.ServiceAccountTokenKey]; ok {
-		// client from token
-		secretConfig := &rest.Config{
-			Host: config.Host,
-			TLSClientConfig: rest.TLSClientConfig{
-				CAData: secret.Data[corev1.ServiceAccountRootCAKey],
-			},
-			BearerToken: string(token),
-		}
-
-		return NewClientSetForConfig(secretConfig, opts)
+		return newClientSetFromToken(config.Host, secret.Data[corev1.ServiceAccountRootCAKey], string(token), opts)
 	}
 
 	return nil, errors.New("no valid kubeconfig found")
 }
 
+func newClientSetFromToken(host string, caData []byte, token string, opts client.Options) (*ClientSet, error) {
```

Comment:
I think this function is not needed. In `NewClientSetFromGoogleSAKey` we also create a config sruct and directly call `NewClientSetForConfig`. A function with a struct parameter is more clear to me and we can create struct inline if we only want a single line of code.

Reply:
makes sense ✅